### ISSUE TITLE
[🐛 Bug]: Selenium Grid 4.11 via Docker fails to start a new session when using a proxy.

### ISSUE THREAD

**Issue by wild-love** [SeleniumHQ/selenium#12657]
### What happened?

My team has been using the Docker version of Selenium grid with chrome/edge/firefox nodes on the same server with a proxy via Javascript for a while. It has been working fine up until grid version 4.11, where we're beginning to see `SessionNotCreatedError` if running tests with the proxy set. If the proxy is not set, or if we downgrade to any version before 4.11, everything works as expected.

### How can we reproduce the issue?

```javascript
const Builder  = require("selenium-webdriver").Builder,
      cucumber = require("@cucumber/cucumber"),
      firefox  = require("selenium-webdriver/firefox"),
      wd_proxy = require("selenium-webdriver/proxy");

/**
 *  Setup a browser with adblock
 *
 *  @async
 *
 *  @param  {object} browser       The browser to set up
 *  @param  {string} adblock_state The browser's adblock state
 *
 *  @return {promise<bool>} The browser was set up
 */
async function setupBrowserWithAdBlock(browser, adblock_state) {
    let caps      = this.getBrowserCapabilities(browser.name, browser.platform),
        builder   = new Builder(),
        grid_url  = process.env.URL || "http://grid_hub:4444",
        proxy_map = {
            "all": process.env.FWD_PROXY_URL_ALL || "fwd_proxy_all:3128",
            "bc":  process.env.FWD_PROXY_URL_BC || "fwd_proxy_bc:3128",
        };

    if (browser.name === "chrome") {
        caps.setLoggingPrefs({ "browser": "INFO", "performance": "INFO" });
    } else if (browser.name === "firefox") {
        caps = new firefox.Options(caps);

        caps.addArguments("--headless");
    } // end if

    builder.withCapabilities(caps);

    if (this.grid_mode) {
        builder.usingServer(`${grid_url}/wd/hub`);
    } // end if

    if (proxy_map.hasOwnProperty(adblock_state)) {
        let proxy_host = proxy_map[adblock_state];
        
        builder.setProxy(wd_proxy.manual({ "http": proxy_host, "https": proxy_host }));
    } // end if

    this.driver = builder.build();

    await this.driver.manage().setTimeouts(this.config.get("client.browser.timeouts"));

    // maximize() is giving inconsistent results; force window to 1920x1080
    return this.driver.manage().window().setRect({ "height": 1080, "width": 1920, "x": 0, "y": 0 });
} // end async function setupBrowserWithAdBlock
```

### Relevant log output

```shell
15:15:46.028 INFO [NodeServer.lambda$createHandlers$2] - Node has been added
15:16:16.170 WARN [DriverServiceSessionFactory.apply] - Error while creating session with the driver service. Unable to obtain: Capabilities {browserName: chrome, browserVersion: 115.0, goog:chromeOptions: {args: [user-agent=Chrome AutoQ/2.12.6]}, goog:loggingPrefs: {browser: INFO, performance: INFO}, pageLoadStrategy: eager, proxy: {httpProxy: fwd_proxy_all:3128, proxyType: manual, sslProxy: fwd_proxy_all:3128}, se:noVncPort: 7900, se:vncEnabled: true}, error class java.util.LinkedHashMap cannot be cast to class org.openqa.selenium.Proxy (java.util.LinkedHashMap is in module java.base of loader 'bootstrap'; org.openqa.selenium.Proxy is in unnamed module of loader org.openqa.selenium.grid.Bootstrap$PossessiveClassLoader @404b9385)
For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors/driver_location/
Build info: version: '4.11.0', revision: '040bc5406b'
System info: os.name: 'Linux', os.arch: 'amd64', os.version: '6.2.0-31-generic', java.version: '11.0.20'
Driver info: driver.version: unknown
15:16:16.172 WARN [SeleniumSpanExporter$1.lambda$export$1] - Error while creating session with the driver service. Unable to obtain: Capabilities {browserName: chrome, browserVersion: 115.0, goog:chromeOptions: {args: [user-agent=Chrome AutoQ/2.12.6]}, goog:loggingPrefs: {browser: INFO, performance: INFO}, pageLoadStrategy: eager, proxy: {httpProxy: fwd_proxy_all:3128, proxyType: manual, sslProxy: fwd_proxy_all:3128}, se:noVncPort: 7900, se:vncEnabled: true}, error class java.util.LinkedHashMap cannot be cast to class org.openqa.selenium.Proxy (java.util.LinkedHashMap is in module java.base of loader 'bootstrap'; org.openqa.selenium.Proxy is in unnamed module of loader org.openqa.selenium.grid.Bootstrap$PossessiveClassLoader @404b9385)
For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors/driver_location/
Build info: version: '4.11.0', revision: '040bc5406b'
System info: os.name: 'Linux', os.arch: 'amd64', os.version: '6.2.0-31-generic', java.version: '11.0.20'
Driver info: driver.version: unknown
15:16:16.173 WARN [SeleniumSpanExporter$1.lambda$export$1] - org.openqa.selenium.remote.NoSuchDriverException: Unable to obtain: Capabilities {browserName: chrome, browserVersion: 115.0, goog:chromeOptions: {args: [user-agent=Chrome AutoQ/2.12.6]}, goog:loggingPrefs: {browser: INFO, performance: INFO}, pageLoadStrategy: eager, proxy: {httpProxy: fwd_proxy_all:3128, proxyType: manual, sslProxy: fwd_proxy_all:3128}, se:noVncPort: 7900, se:vncEnabled: true}, error class java.util.LinkedHashMap cannot be cast to class org.openqa.selenium.Proxy (java.util.LinkedHashMap is in module java.base of loader 'bootstrap'; org.openqa.selenium.Proxy is in unnamed module of loader org.openqa.selenium.grid.Bootstrap$PossessiveClassLoader @404b9385)
For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors/driver_location/
Build info: version: '4.11.0', revision: '040bc5406b'
System info: os.name: 'Linux', os.arch: 'amd64', os.version: '6.2.0-31-generic', java.version: '11.0.20'
Driver info: driver.version: unknown
        at org.openqa.selenium.remote.service.DriverFinder.getPath(DriverFinder.java:25)
        at org.openqa.selenium.remote.service.DriverFinder.getPath(DriverFinder.java:13)
        at org.openqa.selenium.grid.node.config.DriverServiceSessionFactory.apply(DriverServiceSessionFactory.java:142)
        at org.openqa.selenium.grid.node.config.DriverServiceSessionFactory.apply(DriverServiceSessionFactory.java:73)
        at org.openqa.selenium.grid.node.local.SessionSlot.apply(SessionSlot.java:147)
        at org.openqa.selenium.grid.node.local.LocalNode.newSession(LocalNode.java:467)
        at org.openqa.selenium.grid.node.NewNodeSession.execute(NewNodeSession.java:50)
        at org.openqa.selenium.remote.http.Route$TemplatizedRoute.handle(Route.java:193)
        at org.openqa.selenium.remote.http.Route.execute(Route.java:69)
        at org.openqa.selenium.grid.security.RequiresSecretFilter.lambda$apply$0(RequiresSecretFilter.java:62)
        at org.openqa.selenium.remote.tracing.SpanWrappedHttpHandler.execute(SpanWrappedHttpHandler.java:91)
        at org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:63)
        at org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:347)
        at org.openqa.selenium.remote.http.Route.execute(Route.java:69)
        at org.openqa.selenium.grid.node.Node.execute(Node.java:262)
        at org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:347)
        at org.openqa.selenium.remote.http.Route.execute(Route.java:69)
        at org.openqa.selenium.remote.AddWebDriverSpecHeaders.lambda$apply$0(AddWebDriverSpecHeaders.java:35)
        at org.openqa.selenium.remote.ErrorFilter.lambda$apply$0(ErrorFilter.java:44)
        at org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:63)
        at org.openqa.selenium.remote.ErrorFilter.lambda$apply$0(ErrorFilter.java:44)
        at org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:63)
        at org.openqa.selenium.netty.server.SeleniumHandler.lambda$channelRead0$0(SeleniumHandler.java:44)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ClassCastException: class java.util.LinkedHashMap cannot be cast to class org.openqa.selenium.Proxy (java.util.LinkedHashMap is in module java.base of loader 'bootstrap'; org.openqa.selenium.Proxy is in unnamed module of loader org.openqa.selenium.grid.Bootstrap$PossessiveClassLoader @404b9385)
        at org.openqa.selenium.manager.SeleniumManager.getDriverPath(SeleniumManager.java:262)
        at org.openqa.selenium.remote.service.DriverFinder.getPath(DriverFinder.java:22)
        ... 27 more

15:16:16.182 WARN [SeleniumSpanExporter$1.lambda$export$3] - {"traceId": "4716ea52235e1c0984471c1fa680f1e9","eventTime": 1693581376170631782,"eventName": "exception","attributes": {"exception.message": "Error while creating session with the driver service. Unable to obtain: Capabilities {browserName: chrome, browserVersion: 115.0, goog:chromeOptions: {args: [user-agent=Chrome AutoQ\u002f2.12.6]}, goog:loggingPrefs: {browser: INFO, performance: INFO}, pageLoadStrategy: eager, proxy: {httpProxy: fwd_proxy_all:3128, proxyType: manual, sslProxy: fwd_proxy_all:3128}, se:noVncPort: 7900, se:vncEnabled: true}, error class java.util.LinkedHashMap cannot be cast to class org.openqa.selenium.Proxy (java.util.LinkedHashMap is in module java.base of loader 'bootstrap'; org.openqa.selenium.Proxy is in unnamed module of loader org.openqa.selenium.grid.Bootstrap$PossessiveClassLoader @404b9385)\nFor documentation on this error, please visit: https:\u002f\u002fwww.selenium.dev\u002fdocumentation\u002fwebdriver\u002ftroubleshooting\u002ferrors\u002fdriver_location\u002f\nBuild info: version: '4.11.0', revision: '040bc5406b'\nSystem info: os.name: 'Linux', os.arch: 'amd64', os.version: '6.2.0-31-generic', java.version: '11.0.20'\nDriver info: driver.version: unknown","exception.stacktrace": "org.openqa.selenium.remote.NoSuchDriverException: Unable to obtain: Capabilities {browserName: chrome, browserVersion: 115.0, goog:chromeOptions: {args: [user-agent=Chrome AutoQ\u002f2.12.6]}, goog:loggingPrefs: {browser: INFO, performance: INFO}, pageLoadStrategy: eager, proxy: {httpProxy: fwd_proxy_all:3128, proxyType: manual, sslProxy: fwd_proxy_all:3128}, se:noVncPort: 7900, se:vncEnabled: true}, error class java.util.LinkedHashMap cannot be cast to class org.openqa.selenium.Proxy (java.util.LinkedHashMap is in module java.base of loader 'bootstrap'; org.openqa.selenium.Proxy is in unnamed module of loader org.openqa.selenium.grid.Bootstrap$PossessiveClassLoader @404b9385)\nFor documentation on this error, please visit: https:\u002f\u002fwww.selenium.dev\u002fdocumentation\u002fwebdriver\u002ftroubleshooting\u002ferrors\u002fdriver_location\u002f\nBuild info: version: '4.11.0', revision: '040bc5406b'\nSystem info: os.name: 'Linux', os.arch: 'amd64', os.version: '6.2.0-31-generic', java.version: '11.0.20'\nDriver info: driver.version: unknown\n\tat org.openqa.selenium.remote.service.DriverFinder.getPath(DriverFinder.java:25)\n\tat org.openqa.selenium.remote.service.DriverFinder.getPath(DriverFinder.java:13)\n\tat org.openqa.selenium.grid.node.config.DriverServiceSessionFactory.apply(DriverServiceSessionFactory.java:142)\n\tat org.openqa.selenium.grid.node.config.DriverServiceSessionFactory.apply(DriverServiceSessionFactory.java:73)\n\tat org.openqa.selenium.grid.node.local.SessionSlot.apply(SessionSlot.java:147)\n\tat org.openqa.selenium.grid.node.local.LocalNode.newSession(LocalNode.java:467)\n\tat org.openqa.selenium.grid.node.NewNodeSession.execute(NewNodeSession.java:50)\n\tat org.openqa.selenium.remote.http.Route$TemplatizedRoute.handle(Route.java:193)\n\tat org.openqa.selenium.remote.http.Route.execute(Route.java:69)\n\tat org.openqa.selenium.grid.security.RequiresSecretFilter.lambda$apply$0(RequiresSecretFilter.java:62)\n\tat org.openqa.selenium.remote.tracing.SpanWrappedHttpHandler.execute(SpanWrappedHttpHandler.java:91)\n\tat org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:63)\n\tat org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:347)\n\tat org.openqa.selenium.remote.http.Route.execute(Route.java:69)\n\tat org.openqa.selenium.grid.node.Node.execute(Node.java:262)\n\tat org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:347)\n\tat org.openqa.selenium.remote.http.Route.execute(Route.java:69)\n\tat org.openqa.selenium.remote.AddWebDriverSpecHeaders.lambda$apply$0(AddWebDriverSpecHeaders.java:35)\n\tat org.openqa.selenium.remote.ErrorFilter.lambda$apply$0(ErrorFilter.java:44)\n\tat org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:63)\n\tat org.openqa.selenium.remote.ErrorFilter.lambda$apply$0(ErrorFilter.java:44)\n\tat org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:63)\n\tat org.openqa.selenium.netty.server.SeleniumHandler.lambda$channelRead0$0(SeleniumHandler.java:44)\n\tat java.base\u002fjava.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)\n\tat java.base\u002fjava.util.concurrent.FutureTask.run(FutureTask.java:264)\n\tat java.base\u002fjava.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base\u002fjava.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base\u002fjava.lang.Thread.run(Thread.java:829)\nCaused by: java.lang.ClassCastException: class java.util.LinkedHashMap cannot be cast to class org.openqa.selenium.Proxy (java.util.LinkedHashMap is in module java.base of loader 'bootstrap'; org.openqa.selenium.Proxy is in unnamed module of loader org.openqa.selenium.grid.Bootstrap$PossessiveClassLoader @404b9385)\n\tat org.openqa.selenium.manager.SeleniumManager.getDriverPath(SeleniumManager.java:262)\n\tat org.openqa.selenium.remote.service.DriverFinder.getPath(DriverFinder.java:22)\n\t... 27 more\n","exception.type": "org.openqa.selenium.remote.NoSuchDriverException","logger": "org.openqa.selenium.grid.node.config.DriverServiceSessionFactory","session.capabilities": "{\"proxy\": {\"proxyType\": \"manual\",\"httpProxy\": \"fwd_proxy_all:3128\",\"sslProxy\": \"fwd_proxy_all:3128\"},\"goog:loggingPrefs\": {\"browser\": \"INFO\",\"performance\": \"INFO\"},\"goog:chromeOptions\": {\"args\": [\"user-agent=Chrome AutoQ\\u002f2.12.6\"]},\"browserVersion\": \"115.0\",\"se:noVncPort\": 7900,\"browserName\": \"chrome\",\"pageLoadStrategy\": \"eager\",\"se:vncEnabled\": true}\n"}}

15:16:16.183 WARN [SeleniumSpanExporter$1.lambda$export$3] - {"traceId": "4716ea52235e1c0984471c1fa680f1e9","eventTime": 1693581376181992803,"eventName": "Unable to create session with the driver","attributes": {"current.session.count": 0,"logger": "org.openqa.selenium.grid.node.local.LocalNode","session.request.capabilities": "Capabilities {browserName: chrome, goog:chromeOptions: {args: [user-agent=Chrome AutoQ\u002f2.12.6]}, goog:loggingPrefs: {browser: INFO, performance: INFO}, pageLoadStrategy: eager, platformName: linux, proxy: {httpProxy: fwd_proxy_all:3128, proxyType: manual, sslProxy: fwd_proxy_all:3128}}","session.request.downstreamdialect": "[W3C]"}}
```

### Operating System

Ubuntu 22.04.03 and Ubuntu 16.04

### Selenium version

4.11, probably Java, whatever is used with the Docker hub version

### What are the browser(s) and version(s) where you see this issue?

Chrome/Edge/Firefox docker nodes 4.11

### What are the browser driver(s) and version(s) where you see this issue?

Chrome/Edge/Firefox docker nodes 4.11

### Are you using Selenium Grid?

Yes, 4.11

**Comment by titusfortner** [comment]
This should have been fixed in 4.12, please try and let us know.

**Comment by wild-love** [comment]
Can I get 4.12 by just specifying that tag? I dont see it listed in https://hub.docker.com/r/selenium/hub/tags , and using `:4` gets me 4.11. Edit: specifying the 4.12 tag doesn't work, docker cant find the manifests.

**Comment by titusfortner** [comment]
I don't do the docker bits. Not sure the release process. We just released 4.12 bindings/grid last night.

**Comment by ShadowLNC** [comment]
It seems like 4.12 is still missing from DockerHub, not sure if I should raise a separate issue for that?

**Comment by titusfortner** [comment]
There's a Java bug that we want to fix and release 4.12.1.
We were waiting on that to do the docker release.

**Comment by diemol** [comment]
https://github.com/SeleniumHQ/docker-selenium/releases/tag/4.12.1-20230904
